### DISCUSSION TITLE
Debugging overheads

### DISCUSSION THREAD

**Discussion by girlpunk** [QuestPDF/QuestPDF#422] [General]
Does anyone have any info on how much overheads `EnableDebugging` causes? I'd like to enable it in our production builds to give end users a few more details on problems that are (to my knowledge) only available in the Element Trace, however this is only available when debugging is turned on.

**Comment by MarcinZiabek** [comment]
I don't think that this mode has a very significant performance impact. It mostly double the number of elements in the document graph. So if anything, it will hit the GA a little bit harder. But this cost may be not noticeable when the document contains a lot of text or images 😁

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
